### PR TITLE
Make hosts variable name customizable

### PR DESCRIPTION
**Pull request by jan-zmeskal**
This PR enables user of this role to choose a different name for `hosts` variable. The original design of the role assumes that the `hosts` variable holds a list of dictionaries describing RHV hosts. I honestly think that naming the variable `hosts` is not a great design decision in the first place. However, I ran into serious problem with it when using carbon testing framework developed by Red Hat Interoperability QE team. 

Any time carbon executes an Ansible playbook, it defines `hosts` variable as extra_vars on CLI. Whatever is defined in extra_vars has the highest priority in Ansible therefore there's no way how to override. That fack makes this role basically unusable in carbon framework.

Hence the change. The way it is implemented makes it backwards compatible. By default, we still expect the user to provide list of their hosts in `hosts` variable. However newly, they can also specify `hosts_var_name` and override that behaviour. Example:
```yaml
- name: Set up oVirt infrastructure
  hosts: engine

  roles:
    - role: ovirt.infra
      vars:
        data_center_name: Default
        compatibility_version: "4.3"
        rhv_cluster_name: Default
        clusters:
          - name: "{{ rhv_cluster_name }}"
            state: present
            cpu_type: "Intel Westmere Family"
        hosts_var_name: rhv_hosts
        rhv_hosts:
          - name: host_0
            state: present
            address: 1.2.3.4
            password: 123456
            cluster: "{{ rhv_cluster_name }}"
```
In this example, list of hosts is present in `rhv_hosts` variable as opposed to the default `hosts` variable.

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

File: README.md
Comment (mwperina) [inline README.md]: Please don't use rhv in upstream code, let's rename to `ov_hosts` or `ovirt_hosts`

File: roles/ovirt.storages/tasks/main.yml
Comment (mwperina) [inline roles/ovirt.storages/tasks/main.yml]: Hmm, wouldn't it be better to use that approach for the whole infra role? I mean having a pre_task which will define correct hosts fact (for example `ov_ifr_hosts`) and use this fact in all place within the role.

**Review by mwperina — CHANGES_REQUESTED** [review]

**Comment by mwperina** [comment]
@dangel101

**Comment by mwperina** [comment]
ci add to white list

File: roles/ovirt.storages/tasks/main.yml
Comment (jan-zmeskal) [reply roles/ovirt.storages/tasks/main.yml]: That's a nice idea. Let me see what I can do about that.

**Comment by jan-zmeskal** [comment]
@mwperina I implemented the changes suggested by you. I tested it in three basic scenarios:
1. When not providing any hosts
2. When providing hosts in `hosts` variable
3. When providing hosts in `ovirt_hosts` variable

However, I think someone other than me should test this properly before merging as this makes a potentially breaking change at several places

**Review by mwperina — APPROVED** [review]
+1

**Review by mnecas — CHANGES_REQUESTED** [review]
I see the usage of `hosts_var_name` counterintuitive.

Wouldn't it be better to use the variable `ovirt_host` as the second option to which the variable `host` would default to when it would not be specified?
You would need to only add in beggining 
```yaml
- set_facts:
    hosts: "{{ hosts | default(ovirt_hosts) | default(omit) }}"
```
or everywhere use ovirt_hosts and in beginning set it to one of those two vars.

**Comment by jan-zmeskal** [comment]
@mnecas That was actually my original idea but then we would not really solve the problem - we'd only make it much less probable to occur. Which is also good, I admit that. However, we might also run into trouble when the user uses `ovirt_hosts` in their playbook for something else than what we expect. I honestly don't see a big problem in saying: Use `hosts` variable and if you don't like it, use whatever you want. Keep in mind that 99 % of users don't need to be aware of this mechanism at all.

File: roles/ovirt.hosts/tasks/main.yml
Comment (mnecas) [inline roles/ovirt.hosts/tasks/main.yml]: would use if `ovirt_infra_hosts` or `ovirt_infra_hosts is not []` because the `bool` filter only converts string `"True"` to bool

**Review by mnecas — CHANGES_REQUESTED** [review]
That's true, this is really edge case. I don't mind keeping it as it is.
Got just one more comment.

File: roles/ovirt.hosts/tasks/main.yml
Comment (jan-zmeskal) [reply roles/ovirt.hosts/tasks/main.yml]: I implemented the comment but I still think it would be great if we could get somebody else than me to test it before merging

**Review by mwperina — APPROVED** [review]
Looks good to me.

Martine, could you please reverify before merging?

**Review by mnecas — APPROVED** [review]
LGTM
tested on oVirt 4.4, no issues

**Pull request merged**
